### ISSUE TITLE
Display note for "unused variable" warning suggesting that the warning can be disabled via underscores

### ISSUE THREAD

**Issue by nham** [rust-lang/rust#26720]
For reference, see [this Reddit thread](https://www.reddit.com/r/rust/comments/3be26r/for_loop_with_unused_variable/)

It seems to me that if you have code like this

``` rust
for x in foo {
    // code that does not use `x`...
}
```

Then in addition to the "unused variable" warning that you currently get:

```
unused_var.rs:2:9: 2:10 warning: unused variable: `x`, #[warn(unused_variables)] on by default
unused_var.rs:2     for x in [1,2,3].iter() {
                        ^
```

it should also display a note, perhaps saying 

```
note: to disable this warning, consider using `_x` or `_` instead
```

Barring that, the lint could have an error code and the error explanation could mention that you can disable the warning in either of these ways.

I'm not sure if this note should only be displayed with `for` loops (and similar situations), or if it should be unconditionally displayed alongside the "unused variable" warning.

**Comment by steveklabnik** [comment]
Triage: no change.

**Comment by Mark-Simulacrum** [comment]
I'm not sure how to add notes to lints, but in general I think the changes would need to be done somewhere around [here](https://github.com/rust-lang/rust/blob/e0cc22b4bae8007c59fbe58f2c104ecd743d746a/src/librustc/middle/liveness.rs#L1489-L1494). Perhaps someone could write up some instructions?

**Comment by GuillaumeGomez** [comment]
Sounds like fun. Doing it.
